frontend/public/components/modals/alert-routing-modal.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import {Base64} from 'js-base64';
+import {safeDump} from 'js-yaml';
+import * as _ from 'lodash-es';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import {k8sPatch, K8sResourceKind} from '../../module/k8s';
+import {SecretModel} from '../../models';
+
+export const AlertRoutingModal: React.FC<AlertRoutingModalProps> = ({config, secret, cancel, close}) => {
+  const [errorMessage, setErrorMessage] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const submit = (event): void => {
+    event.preventDefault();
+    config = _.cloneDeep(config);
```

Comment:
Should probably assign this to a new variable, rather than overwriting `config`.